test/statistics/stats_test.cpp

```diff
@@ -157,10 +160,10 @@ TEST_F(StatsTests, MultiThreadStatsTest) {
   ASSERT_EQ(aggregated_stats.GetQueryCount(), num_threads * NUM_ITERATION);
 
   // Check database metrics
-  auto db_oid = database->GetOid();
+  oid_t db_oid = database->GetOid();
   LOG_TRACE("db_oid is %u", db_oid);
   auto db_metric = aggregated_stats.GetDatabaseMetric(db_oid);
-  ASSERT_EQ(db_metric->GetTxnCommitted().GetCounter(),
+  ASSERT_GT(db_metric->GetTxnCommitted().GetCounter(),
```

Comment:
What stats test basically does is to start 8 threads, and within each thread invokes method **IncrementTxnCommitted()** 50 times, thus we will eventually have 8 * 50 = 400 committed transaction in that database. But we've changed table/index/query metrics catalog tables from global singleton to per database catalog table, every time we insert a stats record into corresponding catalog table, the number of committed transaction will be plus one. Since we don't have full control over how long the test will run and how many times the stats aggregator thread will be woken up, we can only have a nondeterministic number of committed transaction(but always larger than 400).

Reply:
Actually, could you add this explanation as a comment? So if someone changes the test, they know what they should expect.

Reply:
Sure!